src/missing.c

```diff
@@ -148,64 +156,57 @@ void col_equal_na(r_obj* x,
 
 #define COL_EQUAL_NA(CTYPE, CBEGIN, IS_MISSING) do { \
```

Comment:
Or like `rlang::flatten_if(x, predicate)`, it could be `locate_if(x, predicate)`.

Locate has been our indicator that we are returning a vector of locations.

Maybe something like:

```c
r_obj* r_int_locate_if(int* v_x, bool (predicate)(int))
```

Seems a bit hard to generalize to the data frame case though.

I'll merge as is and keep in mind if I see this pattern again